src/main/java/chess/dao/converter/StateConverter.java

```diff
@@ -0,0 +1,48 @@
+package chess.dao.converter;
+
+import chess.model.board.Board;
+import chess.model.state.State;
+import chess.model.state.finished.End;
+import chess.model.state.running.BlackTurn;
+import chess.model.state.running.WhiteTurn;
+import java.util.Arrays;
+
+public enum StateConverter {
+
+    WHITE_TURN("WHITE_TURN"),
```

Comment:
```suggestion
    WHITE_TURN("WHITE_TURN", new WhiteTurn(board)),
```

이런 구조로 만들면 createState 메서드의 if를 없앨 수 있겠군요